### PR TITLE
Hint/cleanup/20190719

### PR DESCRIPTION
**Pull request by dinjas**
Performs a lot of cleanup and begins addressing rubocop issues.

### CODE REVIEW

File: app/lib/services/notifications/needs/message/destroy.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+                   to: :need
+
+          def message
+            "A need at #{office} has been deleted.".freeze
+          end
+        end
+      end
+    end
+  end
+end
```
Comment (kyleboe) [inline app/lib/services/notifications/needs/message/destroy.rb]: Is this file supposed to be in this PR?

File: app/lib/services/notifications/needs/create.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        end
+
+        def shift_users
+          (users_to_notify | shifts.flat_map(&:users_to_notify)).uniq
+        end
+
+      end
+    end
+  end
+end
```
Comment (kyleboe) [inline app/lib/services/notifications/needs/create.rb]: Is this file supposed to be in this PR?

File: spec/lib/includers/shift_update_event_helper_spec.rb [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+    end
+
+    it 'returns empty collection if no social workers' do
+      result = object.social_workers
+
+      expect(result).to match_array([])
+    end
+  end
+
+end
```
Comment (kyleboe) [inline spec/lib/includers/shift_update_event_helper_spec.rb]: See comment above

File: spec/lib/includers/start_at_helper_spec.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+  describe '#start_time' do
+    it 'start_time' do
+      result = shift.start_time
+
+      expect(result).to eql('08:23am')
+    end
+  end
+
+end
```
Comment (kyleboe) [inline spec/lib/includers/start_at_helper_spec.rb]: Is this file supposed to be in this PR?

File: spec/lib/services/notifications/needs/message_spec.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+      it 'returns expected message for destroy event' do
+        result = object.message
+
+        expect(result).to eql("A need at #{need.office} has been deleted.")
+      end
+    end
+  end
+
+end
```
Comment (kyleboe) [inline spec/lib/services/notifications/needs/message_spec.rb]: Is this file supposed to be in this PR?

File: spec/lib/services/notifications/needs/message/update_spec.rb [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+    it 'returns expected message for update event' do
+      result = object.message
+
+      expect(result)
+        .to eql("A new need starting #{day_str} at #{time_str} has opened "\
+                "up at your local office! #{url}")
+    end
+  end
+end
```
Comment (kyleboe) [inline spec/lib/services/notifications/needs/message/update_spec.rb]: Is this file supposed to be in this PR?

File: spec/lib/services/notifications/shifts/message/destroy_spec.rb [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    it 'returns expected string' do
+      result = object
+
+      expect(result).to eql('The shift from 11:15am to 01:15pm has been '\
+                              'removed from a need at your local office. '\
+                              'http://localhost:3000/needs')
+      expect(result).to be_frozen
+    end
+  end
+end
```
Comment (kyleboe) [inline spec/lib/services/notifications/shifts/message/destroy_spec.rb]: Is this file supposed to be in this PR?

**Comment by dinjas** [comment]
still very much WIP. I'll clean it up. :)

**Comment by kyleboe** [comment]
Ah sorry, didn't mean to jump the gun.

**Comment by dinjas** [comment]
Np! Should have made it a draft :)

**Comment by dinjas** [comment]
@kyleboe ready for review when time permits!

**Pull request merged**